src/message/MessageListWeb.js

```diff
@@ -39,15 +41,47 @@ export default class MessageListWeb extends PureComponent<Props> {
     webViewEventHandlers[handler](this.props, eventData);
   };
 
+  getHtml = (props: Props) => {
+    const { auth } = props;
+
+    const messageAsHtml = renderMessagesAsHtml(props);
+    return messageAsHtml.join('').replace(/src="\//g, `src="${auth.realm}/`);
+  };
+
+  sendMessage = (msg: Object) => {
+    if (this.webview) {
+      this.webview.postMessage(JSON.stringify(msg), '*');
+    }
+  };
+
+  shouldComponentUpdate(nextProps: Props) {
+    if (this.props.fetchingOlder !== nextProps.fetchingOlder) {
+      // toggle top loading-spinner
+      this.sendMessage({
+        type: 'loading-top',
+        newState: nextProps.fetchingOlder,
+      });
+      return false;
+    }
+    if (nextProps.fetchingNewer !== this.props.fetchingNewer) {
+      this.sendMessage({
+        type: 'loading-bottom',
+        newState: nextProps.fetchingNewer,
+      });
+      return false;
+    }
+
+    return shallowCompare(this, nextProps);
```

Comment:
This is not even true :) 
You should compare this.props to nextProps (but also no need since you will move all the logic to `componentWillReceiveProps`.